private/buf/bufformat/formatter.go

```diff
@@ -2048,9 +2067,45 @@ func (f *formatter) writeComment(comment string) {
 				line = unindent(line, minIndent)
 				line = strings.TrimRightFunc(line, unicode.IsSpace)
 			}
+			// If we have a block comment with no prefix, we'll format
+			// like so:
+
+			/*
+			   This is a multi-line comment example.
+			   It has no comment prefix on each line.
+			*/
+
+			// But if there IS a prefix, "|" for example, we'll left-align
+			// the prefix symbol under the asterisk of the comment start
+			// like this:
+
+			/*
+			 | This comment has a prefix before each line.
+			 | Usually the prefix is asterisk, but it's a
+			 | pipe in this example.
+			*/
```

Comment:
I had considered lining up the asterisk of the trailing `*/` so it is left-aligned underneath the other symbols. _However_, I couldn't even put an example like that in here because `gofmt` would unindent the `*/` by one space.

So this alignment of the `*/` terminator is inspired by `gofmt`. Its rules appear to be that the `*/` is exactly aligned with the `/*` _unless_ each comment line starts with an asterisk, in which case it is aligned exactly one space to the right of the `/*` so that the asterisks all line up.